libraries/core_libs/network/src/tarcap/packets_handlers/dag_block_packet_handler.cpp

```diff
@@ -63,6 +63,7 @@ void DagBlockPacketHandler::process(const PacketData &packet_data, const std::sh
           LOG(log_er_) << "DagBlock" << block.getHash() << " has missing pivot or/and tips " << status.second
                        << " . Peer " << packet_data.from_node_id_ << " will be disconnected.";
           disconnect(peer->getId(), dev::p2p::UserReason);
+          peers_state_->set_peer_malicious(peer->getId());
```

Comment:
return ?

Reply:
lines 63-66 should be replaced by:

`throw MaliciousPeerException(err_msg);`